### DISCUSSION TITLE
Support a custom HTTP header for the source of requests

### DISCUSSION THREAD

**Discussion by joeblubaugh** [grafana/grafana#68049] [General]
Grafana APIs (and plugin APIs) may be called by various clients:
- terraform providers
- web UI
- custom client code
- github.com/K-phoen/grabana
- github.com/grafana/grizzly

It could be useful to distinguish these sources. How about adding an optional custom HTTP header that the backend can recognize? One use case: distinguishing configuration provided by as-code tools could let us develop features to protect them from manual edits if a user configures Grafana to do so.

**Comment by joeblubaugh** [comment]
A few possibilities: `Grafana-Client-Type`, `Grafana-Request-Source`?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82697. Feel free to continue the discussion around this there. Thank you!
